src/components/tag/Tag.root.tsx

```diff
@@ -0,0 +1,60 @@
+import styled, { css } from 'styled-components';
+import { FlattenSimpleInterpolation, ThemedStyledProps } from 'styled-components/ts3.6';
+
+import { Spacing } from '../../foundations/layout';
+import { applyPadding } from '../../foundations/layout/Grid.mixins';
+import { applyShape } from '../../foundations/shape/Shape.mixins';
+import { TTheme } from '../../foundations/theme-provider/themes/theme.types';
+import { applyTextMargin, applyTextStyles } from '../text/Text.mixins';
+import Text from '../text';
+import { Utils } from '../../shared';
+
+import { PTagRoot } from './Tag.props';
+import { TTagVariant } from './Tag.types';
+
+const TagRoot = styled(Text).withConfig({
+    shouldForwardProp: Utils.forwardProperties(),
+})<PTagRoot>(
+    ({
+        variant,
+        size,
+        onClick,
```

Comment:
nit: it would be probably good to send a boolean instead of the function over, since callbacks can potentially be really heavy I would like to avoid that as much as possible.

You can just pass in a prop like `isClickable` or something like that and move the condition from below to that prop definition.

Reply:
that's a great idea actually, thank you!